src/Log/LogService.php

```diff
@@ -392,35 +392,66 @@ public function setDefaultFormat($format)
         $this->defaultFormat = $format;
     }
 
+//    /**
+//     * Returns the default log output format.
+//     *
+//     * The output format can be set explicitly in this class or
+//     * be configured in the global configuration.
+//     *
+//     * A run ID is added to the output format if the placeholder %runId%
+//     * is present in order to match log output to separate requests.
+//     *
+//     * @return string
+//     */
+//    public function getDefaultFormat()
+//    {
+//        if ($this->defaultFormat == null) {
+//            $config = $this->getConfig();
+//
+//            $format = self::DEFAULT_FORMAT;
+//
+//            if (isset($config->log->format)) {
+//                $format = $config->log->format;
+//            }
+//
+//            $this->defaultFormat = $format;
+//        }
+//
+//        $runId = $this->getRunId();
+//
+//        $format = rtrim(preg_replace('/%runId%/', $runId, $this->defaultFormat), PHP_EOL) . PHP_EOL;
+//
+//        return $format;
+//    }
+
     /**
-     * Returns the default log output format.
-     *
-     * The output format can be set explicitly in this class or
-     * be configured in the global configuration.
-     *
-     * A run ID is added to the output format if the placeholder %runId%
-     * is present in order to match log output to separate requests.
+     * Returns either custom log format or default format if custom log format doesn't exist.
      *
+     * @param $logName string
      * @return string
      */
-    public function getDefaultFormat()
+    public function getLogFormat($logName)
```

Comment:
Why should `getLogFormat` replace `getDefaultFormat`? The old function has a specific reason to exist. However it has a problem. It does two jobs. It gets the default format and replaces %runId%. The new function does three jobs. The two of `getDefaultFormat` and part of `getLogConfig`. That is not improving things. It also does not solve the remaining problem, that `createLog` can be called with a format and that this string might contain %runId%. That case is still not handled. We need two functions, one is the old `getDefaultFormat` with almost no modifications and the new one needs to take care of replacing %runId% in the format, no matter where it is coming from.

Please don't use `//` to comment out code blocks and in this case here, just delete the code. We want to avoid unnecessary turnarounds and even if everything would be meeting the requirements I would have to bounce it back to get it cleaned up  before I could accept and merge the PR.